src/views/IdentityOverview/IdentityOverview.tsx

```diff
@@ -45,6 +45,22 @@ export function IdentityOverview({ identity }: Props): JSX.Element | null {
   const showDownloadPrompt =
     credentials && credentials.some(({ isDownloaded }) => !isDownloaded);
 
+  const [hasLegacyDid, setHasLegacyDid] = useState(false);
+
+  useEffect(() => {
+    (async () => {
+      if (await needLegacyDidCrypto(identity.did)) {
+        setHasLegacyDid(true);
+      } else {
+        setHasLegacyDid(false);
```

Comment:
It is already false by default, right?
```suggestion
```

Reply:
I initially had the same thought, but leaving out this line causes the 'true' state to persist when the identity is changed with the arrow buttons.